### ISSUE TITLE
Integer column will change to bigint type after running db:migrate in Rails 6.1 and SQLite3

### ISSUE THREAD

**Issue by JunichiIto** [rails/rails#43168]
### Steps to reproduce

```
git clone git@github.com:JunichiIto/references-sandbox.git
bundle install
rails db:migrate
git diff
```

Then, you will see:

```diff
diff --git a/db/schema.rb b/db/schema.rb
index 8e88a0f..be7c552 100644
--- a/db/schema.rb
+++ b/db/schema.rb
@@ -13,7 +13,7 @@
 ActiveRecord::Schema.define(version: 2021_09_05_003200) do
 
   create_table "books", force: :cascade do |t|
-    t.integer "user_id", null: false
+    t.bigint "user_id", null: false
     t.string "title"
     t.datetime "created_at", precision: 6, null: false
     t.datetime "updated_at", precision: 6, null: false
```

### Expected behavior

`git diff` should show nothing. (no difference)

### Actual behavior

`user_id` type changed from integer to bigint

### System configuration

- Created and ran the CreateBooks migration in Rails 6.0.4.1, then upgraded to 6.1.4.1
- RDBMS: SQLite3

**Ruby version**: 3.0.0

### Other info

I understand I should run `db:schema:load` instead of `db:migrate`, but they should behave in the same way.

I guess the following commit by @lukeredpath is related to this issue:

https://github.com/rails/rails/commit/e8437a68f614d61722d94445a9792ccde5dfa0da#diff-01b7232fa00908de002595608171bf83348e1898a24fb2af88d7d44175bacb63R26-R30

The code above might override the default type (integer) in SQLite3:

https://github.com/rails/rails/blob/v6.1.4.1/activerecord/lib/active_record/connection_adapters/sqlite3/schema_definitions.rb#L7-L9

And I confirmed the type won't change when I change the Migration version from 6.0 to 6.1:

```diff
-class CreateBooks < ActiveRecord::Migration[6.0]
+class CreateBooks < ActiveRecord::Migration[6.1]
```

I found this issue when I ran the CreateActiveStorageVariantRecords migration:

```ruby
# frozen_string_literal: true

# This migration comes from active_storage (originally 20191206030411)
# rubocop:disable all
class CreateActiveStorageVariantRecords < ActiveRecord::Migration[6.0]
  def change
    create_table :active_storage_variant_records do |t|
      t.belongs_to :blob, null: false, index: false
      t.string :variation_digest, null: false

      t.index %i[ blob_id variation_digest ], name: "index_active_storage_variant_records_uniqueness", unique: true
      t.foreign_key :active_storage_blobs, column: :blob_id
    end
  end
end
```

It is automatically created after running `rails app:update` in Rails 6.1. After running `rails db:migrate`, `blob_id` has bigint type, but it should be integer in SQLite3.

**Comment by ghiculescu** [comment]
Thanks for reporting this.

Is `bigint` a valid type in SQLite or is it not supported at all?

Based on this report, it seems like `t.references` and `t.belongs_to` would always be on SQLite since Rails 6.1. If so I'm surprised it hasn't been reported more often.

It looks like `ReferenceDefinition.new` defaults to `bigint` here: https://github.com/rails/rails/blob/18707ab17fa492eb25ad2e8f9818a320dc20b823/activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb#L155

Maybe it would be better if instead a default value being used, a method was called on the default adapter. That would probably work better than the [overriding](https://github.com/rails/rails/blob/v6.1.4.1/activerecord/lib/active_record/connection_adapters/sqlite3/schema_definitions.rb#L7-L9) approach which has been overridden again.

**Comment by JunichiIto** [comment]
@ghiculescu Thank you for your reply.

> Is `bigint` a valid type in SQLite or is it not supported at all?

`bigint` is a valid type. Please refer "3.1.1. Affinity Name Examples" in the page below:
https://sqlite.org/datatype3.html

But `bigint` behaves as same as `integer`, so no error happens. My concern is `schema.rb` gets a tiny diff after running `rails app:update` and `rails db:migrate` in Rails 6.1.

**Comment by ghiculescu** [comment]
Hmm, yes, that is a bit annoying. I think that could be fixed (see above comment). Would you be interested in making a PR?

**Comment by marcelolx** [comment]
> Based on this report, it seems like `t.references` and `t.belongs_to` would always be wrong on SQLite since Rails 6.1. If so I'm surprised it hasn't been reported more often.

@ghiculescu I was looking at this one and as far as I can tell this problem happens only for migrations with `6.0` version as  @JunichiIto mentions in his report ⬇️ 
> And I confirmed the type won't change when I change the Migration version from 6.0 to 6.1

With Rails >= 6.1 the type is passed to `add_reference` here https://github.com/rails/rails/blob/main/activerecord/lib/active_record/connection_adapters/sqlite3_adapter.rb#L304 and in the case of `t.references`/`t.belongs_to` in the table definition override that exists for SQLite https://github.com/rails/rails/blob/main/activerecord/lib/active_record/connection_adapters/sqlite3/schema_definitions.rb#L8. In fact, the same way it always was and this didn't change in that PR. 

I think that the reason that this hasn't been reported more often is that since the problem is in the compatibility module for `6.0` versions of migrations it doesn't even happened to most of the users.

I opened a PR to address this https://github.com/rails/rails/pull/43295

**Comment by JunichiIto** [comment]
@marcelolx Thank you for you PR. I confirmed the issue is fixed by https://github.com/rails/rails/pull/43295 😃

**Comment by gazayas** [comment]
Having the same issue here:
### System Configuration
Rails: 7.0.2
PostgreSQL: 13
Ruby: 3.1.2

The problem happens when running `rails app:update` and upgrading to 7.0.3. Since `ActiveRecord` in the schema only shows up to minor versions, `ActiveRecord::Schema[7.0]` is left unchanged but foreign key ints are being changed to bigints for me.

I saw the [changelog](https://github.com/rails/rails/blob/v7.0.3/activerecord/CHANGELOG.md) merged in #43295, so I figured this was related.
